### PR TITLE
Mac clang build

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -55,9 +55,11 @@ CFLAGS = \
 
 # Compile flags for linux / osx
 ifeq ($(OS),linux)
+CFLAGS += -fopenmp
 SHOBJ_LDFLAGS ?= -Wl,-Bsymbolic,-Bsymbolic-functions -fopenmp -shared -ldl -lpthread
 else
-SHOBJ_LDFLAGS ?= -bundle -undefined dynamic_lookup -ldl -lpthread -fopenmp
+CFLAGS += -Xpreprocessor -fopenmp
+SHOBJ_LDFLAGS ?= -bundle -undefined dynamic_lookup -ldl -lpthread -Xpreprocessor -fopenmp -lomp
```
Comment (swilly22) [inline src/Makefile:62]: not sure if we need both `-fopenmp -lomp`
do we need openmp flag on both `CFLAGS` and `SHOBJ_LDFLAGS` ?

**Review by swilly22 — COMMENTED** [review]
@AviAvni please see comment

File: .circleci/config.yml [hunk 2]
```diff
@@ -299,22 +299,22 @@ jobs:
...
             - deps/GraphBLAS/build
-          key: GraphBLAS-mac-{{ checksum "deps/GraphBLAS/README.md" }}
+          key: GraphBLAS-mac-clang-{{ checksum "deps/GraphBLAS/README.md" }}
       - run:
           name: Test
           shell: /bin/bash -l -eo pipefail
           command: |
             ulimit -n 10000
-            make test
+            DYLD_LIBRARY_PATH=/usr/local/opt/libomp/lib CC=clang CXX=clang++ make test
```
Comment (chayim) [inline .circleci/config.yml:317]: Can we use the direct /usr/local/lib here? Brew should be installing its taps there.

**Review by rafie — APPROVED** [review]

File: deps/GraphBLAS/README.md [hunk 3]
```diff
@@ -218,3 +218,4 @@ BSD-2-clause license.
 Intel (also Apache 2.0), and the cpu_featurer/ndk_compat component (c)
 The Android Open Source Project (BSD-2-clause)
 
+CLEAR CACHE
```
Comment (swilly22) [inline deps/GraphBLAS/README.md:221]: can't we force CI clear cache via the CI platform ?

**Review by swilly22 — COMMENTED** [review]
@AviAvni please see single comment

File: deps/GraphBLAS/README.md (hunk 3 above)
Comment (AviAvni) [reply deps/GraphBLAS/README.md:221]: Unfortunately no or you change the key or like in our case change the README file

Comment (swilly22) [reply deps/GraphBLAS/README.md:221]: fuggedaboutit
https://circleci.com/docs/2.0/caching/#clearing-cache

**Review by swilly22 — APPROVED** [review]

**Pull request merged**